components/skus/browser/skus_context_impl.cc

```diff
@@ -133,36 +172,99 @@ std::unique_ptr<SkusUrlLoader> shim_executeRequest(
 
 SkusContextImpl::SkusContextImpl(
     PrefService* prefs,
-    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
-    : prefs_(*prefs), url_loader_factory_(url_loader_factory) {}
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+    scoped_refptr<base::SingleThreadTaskRunner> sdk_task_runner,
+    scoped_refptr<base::SequencedTaskRunner> ui_task_runner)
+    : prefs_(*prefs),
+      sdk_task_runner_(sdk_task_runner),
+      ui_task_runner_(ui_task_runner),
+      url_loader_factory_(url_loader_factory) {}
 
 SkusContextImpl::~SkusContextImpl() = default;
 
 std::unique_ptr<skus::SkusUrlLoader> SkusContextImpl::CreateFetcher() const {
   return std::make_unique<SkusUrlLoaderImpl>(url_loader_factory_);
 }
 
-std::string SkusContextImpl::GetValueFromStore(std::string key) const {
+void SkusContextImpl::ScheduleGetValueFromStore(
+    std::string key,
+    rust::cxxbridge1::Fn<void(rust::cxxbridge1::Box<skus::StorageGetContext>,
+                              rust::String value,
+                              bool success)> done,
+    rust::cxxbridge1::Box<skus::StorageGetContext> st_ctx) const {
+  VLOG(1) << "shim_get: `" << key << "`";
+  ui_task_runner_->PostTask(FROM_HERE,
```

Comment:
this is not safe for the same reason that the calls in SkusServiceImpl are not safe, but additionally because the PrefService may be destroyed before the sdk task runner